app/views/locations/upload_locations_csv.html.erb

```diff
@@ -0,0 +1,95 @@
+<h1 class="govuk-heading-l">Upload Summary</h1>
+
+<% if @parent_organisation.errors.present? %>
+  <div class="govuk-error-summary" role="alert" data-module="govuk-error-summary" aria-labelledby="error-summary-title">
+    <h2 id="error-summary-title" class="govuk-error-summary__title">
+      There is a problem
+    </h2>
+    <div class="govuk-error-summary__body">
+      <ul class="govuk-list govuk-error-summary__list">
+        <% @parent_organisation.locations.select(&:new_record?).each_with_index do |location, i| %>
+          <% (location.errors.where(:address) + location.errors.where(:postcode)).first&.tap do |error| %>
+            <li>
+              <a href='#<%= upload_error_id_helper(i, location.address) %>'><%= error.full_message %></a>
+            </li>
+          <% end %>
+          <% location.ips.each do |ip| %>
+              <% ip.errors.first&.tap do |error| %>
+              <li>
+                <a href='#<%= upload_error_id_helper(i, location.address, ip.address) %>'>
+                  <%= error.full_message %>
+                </a>
+              </li>
+            <% end %>
+          <% end %>
+        <% end %>
+      </ul>
+    </div>
+  </div>
+<% end %>
+
+<% @parent_organisation.locations.select(&:new_record?).each_with_index do |location, i| %>
+
+<fieldset class="govuk-fieldset">
+  <legend class="govuk-fieldset__legend govuk-fieldset__legend--m">
+    <% location_errors = (location.errors.where(:address) + location.errors.where(:postcode)) %>
+    <% if location_errors.any? %>
+      <div class="govuk-form-group govuk-form-group--error">
+        <% location_errors.first.tap do |error| %>
```

Comment:
I think this could be simplified by doing `&.tap`. Then you wouldn't need the `.any?` check.